[CI] JvmCrashIT testJvmCrash failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #3146 / windows-2019_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/cniuylvdk2rxs)
- [elasticsearch-periodic-platform-support #3146 / windows-2022_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/cmqcguq3vfnym)
- [elasticsearch-periodic-platform-support #3146 / windows-2016_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/7mxwobmpyuqq2)

**Reproduction Line:**
```
gradlew ':test:external-modules:test-jvm-crash:javaRestTest' --tests "org.elasticsearch.test.jvm_crash.JvmCrashIT.testJvmCrash" -Dtests.seed=BCB1C54884DD6ACB -Dtests.locale=es-HN -Dtests.timezone=UCT -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testJvmCrash'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.test.jvm_crash.JvmCrashIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: with toString() a string matching the pattern <.*# A fatal error has been detected by the Java Runtime Environment:.*SIGSEGV.*>
     but: toString() the string was "[2024-06-21T14:03:42,879][INFO ][o.e.n.NativeAccess       ] [test-cluster-0] Using [jdk] native provider and native methods for [Windows]\r\n[2024-06-21T14:03:43,629][INFO ][o.a.l.i.v.PanamaVectorizationProvider] [test-cluster-0] Java vector incubator API enabled; uses preferredBitSize=256; FMA enabled\r\n[2024-06-21T14:03:44,642][INFO ][o.e.n.Node               ] [test-cluster-0] version[8.15.0-SNAPSHOT], pid[18612], build[zip/95d855923d17c4c64fd6a49d53c47930c87d06c9/2024-06-21T12:10:58.761630400Z], OS[Windows Server 2016/10.0/amd64], JVM[Oracle Corporation/OpenJDK 64-Bit Server VM/22.0.1/22.0.1+8-16]\r\n[2024-06-21T14:03:44,647][INFO ][o.e.n.Node               ] [test-cluster-0] JVM home [C:\\Users\\buildkite\\.gradle\\jdks\\oracle_corporation-22-amd64-windows\\jdk-22.0.1], using bundled JDK [false]\r\n[2024-06-21T14:03:44,648][INFO ][o.e.n.Node               ] [test-cluster-0] JVM arguments [-Des.networkaddress.cache.ttl=60, -Des.networkaddress.cache.negative.ttl=10, -Djava.security.manager=allow, -XX:+AlwaysPreTouch, -Xss1m, -Djava.awt.headless=true, -Dfile.encoding=UTF-8, -Djna.nosys=true, -XX:-OmitStackTraceInFastThrow, -Dio.netty.noUnsafe=true, -Dio.netty.noKeySetOptimization=true, -Dio.netty.recycler.maxCapacityPerThread=0, -Dlog4j.shutdownHookEnabled=false, -Dlog4j2.disable.jmx=true, -Dlog4j2.formatMsgNoLookups=true, -Djava.locale.providers=SPI,COMPAT, --add-opens=java.base/java.io=org.elasticsearch.preallocate, --enable-native-access=org.elasticsearch.nativeaccess,org.apache.lucene.core, -XX:ReplayDataFile=logs/replay_pid%p.log, -Djava.library.path=C:\\bk\\test\\external-modules\\jvm-crash\\build\\testrun\\javaRestTest\\temp\\test-cluster3175310756303214425\\test-cluster-0\\distro\\8.15.0\\lib\\platform\\windows-x64;C:\\Users\\buildkite\\.gradle\\jdks\\oracle_corporation-22-amd64-windows\\jdk-22.0.1\\bin;C:\\Windows\\Sun\\Java\\bin;C:\\Windows\\system32;C:\\Windows;., -Djna.library.path=C:\\bk\\test\\external-modules\\jvm-crash\\build\\testrun\\javaRestTest\\temp\\test-cluster3175310756303214425\\test-cluster-0\\distro\\8.15.0\\lib\\platform\\windows-x64;C:\\Users\\buildkite\\.gradle\\jdks\\oracle_corporation-22-amd64-windows\\jdk-22.0.1\\bin;C:\\Windows\\Sun\\Java\\bin;C:\\Windows\\system32;C:\\Windows;., -Des.distribution.type=zip, -XX:+UnlockDiagnosticVMOptions, -XX:G1NumCollectionsKeepPinned=10000000, -XX:+UseG1GC, -Djava.io.tmpdir=C:\\bk\\test\\external-modules\\jvm-crash\\build\\testrun\\javaRestTest\\temp\\test-cluster3175310756303214425\\test-cluster-0\\tmp, --add-modules=jdk.incubator.vector, -XX:+HeapDumpOnOutOfMemoryError, -XX:+ExitOnOutOfMemoryError, -XX:HeapDumpPath=C:\\bk\\test\\external-modules\\jvm-crash\\build\\testrun\\javaRestTest\\temp\\test-cluster3175310756303214425\\test-cluster-0\\logs, -XX:ErrorFile=C:\\bk\\test\\external-modules\\jvm-crash\\build\\testrun\\javaRestTest\\temp\\test-cluster3175310756303214425\\test-cluster-0\\logs\\hs_err_pid%p.log, -Xlog:gc*,gc+age=trace,safepoint:file=C:\\bk\\test\\external-modules\\jvm-crash\\build\\testrun\\javaRestTest\\temp\\test-cluster3175310756303214425\\test-cluster-0\\logs\\gc.log:utctime,level,pid,tags:filecount=32,filesize=64m, -Xms512m, -Xmx512m, -ea, -esa, -Djvm.crash=true, -XX:MaxDirectMemorySize=268435456, -XX:G1HeapRegionSize=4m, -XX:InitiatingHeapOccupancyPercent=30, -XX:G1ReservePercent=15, --module-path=C:\\bk\\test\\external-modules\\jvm-crash\\build\\testrun\\javaRestTest\\temp\\test-cluster3175310756303214425\\test-cluster-0\\distro\\8.15.0\\lib, --add-modules=jdk.net, --add-modules=ALL-MODULE-PATH, -Djdk.module.main=org.elasticsearch.server]\r\n[2024-06-21T14:03:44,649][INFO ][o.e.n.Node               ] [test-cluster-0] Default Locale [en_US]\r\n[2024-06-21T14:03:44,650][WARN ][o.e.n.Node               ] [test-cluster-0] version [8.15.0-SNAPSHOT] is a pre-release version of Elasticsearch and is not suitable for production\r\n[2024-06-21T14:03:45,911][INFO ][o.e.p.PluginsService     ] [test-cluster-0] loaded module [x-pack-core]\r\n[2024-06-21T14:03:45,911][INFO ][o.e.p.PluginsService     ] [test-cluster-0] loaded module [x-pack-security]\r\n[2024-06-21T14:03:45,912][INFO ][o.e.p.PluginsService     ] [test-cluster-0] loaded module [transport-netty4]\r\n[2024-06-21T14:03:45,912][INFO ][o.e.p.PluginsService     ] [test-cluster-0] loaded module [test-jvm-crash]\r\n[2024-06-21T14:03:47,152][INFO ][o.e.c.u.FeatureFlag      ] [test-cluster-0] The current build is a snapshot, feature flag [madv_random] is enabled\r\n[2024-06-21T14:03:47,522][INFO ][o.e.e.NodeEnvironment    ] [test-cluster-0] using [1] data paths, mounts [[(C:)]], net usable_space [287.7gb], net total_space [349.8gb], types [NTFS]\r\n[2024-06-21T14:03:47,523][INFO ][o.e.e.NodeEnvironment    ] [test-cluster-0] heap size [512mb], compressed ordinary object pointers [true]\r\n[2024-06-21T14:03:47,543][INFO ][o.e.n.Node               ] [test-cluster-0] node name [test-cluster-0], node ID [H7cMutpyT02cCctNxO4BKA], cluster name [test-cluster], roles [data_cold, data, remote_cluster_client, master, data_warm, data_content, transform, data_hot, ml, data_frozen, ingest]\r\n[2024-06-21T14:03:48,462][INFO ][o.e.i.r.RecoverySettings ] [test-cluster-0] using rate limit [40mb] with [default=40mb, read=0b, write=0b, max=0b]\r\n[2024-06-21T14:03:48,659][INFO ][o.e.f.FeatureService     ] [test-cluster-0] Registered local node features [desired_node.version_deprecated, features_supported, file_settings, health.extended_repository_indicator, knn_retriever_supported, license-trial-independent-version, mapper.pass_through_priority, mapper.range.null_values_off_by_one_fix, mapper.source.synthetic_source_fallback, mapper.track_ignored_source, mapper.vectors.int4_quantization, rest.capabilities_action, retrievers_supported, script.hamming, security.migration_framework, security.roles_metadata_flattened, standard_retriever_supported, stats.include_disk_thresholds, unified_highlighter_matched_fields, usage.data_tiers.precalculate_stats]\r\n[2024-06-21T14:03:48,995][INFO ][o.e.x.s.Security         ] [test-cluster-0] Security is disabled\r\n[2024-06-21T14:03:49,288][INFO ][o.e.t.n.NettyAllocator   ] [test-cluster-0] creating NettyAllocator with the following configs: [name=unpooled, suggested_max_allocation_size=1mb, factors={es.unsafe.use_unpooled_allocator=null, g1gc_enabled=true, g1gc_region_size=4mb, heap_size=512mb}]\r\n[2024-06-21T14:03:49,372][INFO ][o.e.d.DiscoveryModule    ] [test-cluster-0] using discovery type [multi-node] and seed hosts providers [settings, file]\r\n[2024-06-21T14:03:50,569][INFO ][o.e.n.Node               ] [test-cluster-0] initialized\r\n[2024-06-21T14:03:50,571][INFO ][o.e.n.Node               ] [test-cluster-0] starting ...\r\n[2024-06-21T14:03:50,765][INFO ][o.e.t.TransportService   ] [test-cluster-0] publish_address {127.0.0.1:59794}, bound_addresses {[::1]:59793}, {127.0.0.1:59794}\r\n[2024-06-21T14:03:51,039][INFO ][o.e.c.c.ClusterBootstrapService] [test-cluster-0] this node has not joined a bootstrapped cluster yet; [cluster.initial_master_nodes] is set to [test-cluster-0]\r\n[2024-06-21T14:03:51,046][WARN ][o.e.d.FileBasedSeedHostsProvider] [test-cluster-0] expected, but did not find, a dynamic hosts list at [C:\\bk\\test\\external-modules\\jvm-crash\\build\\testrun\\javaRestTest\\temp\\test-cluster3175310756303214425\\test-cluster-0\\config\\unicast_hosts.txt]\r\n[2024-06-21T14:03:51,046][INFO ][o.e.c.c.Coordinator      ] [test-cluster-0] setting initial configuration to VotingConfiguration{H7cMutpyT02cCctNxO4BKA}\r\n[2024-06-21T14:03:51,143][INFO ][o.e.h.AbstractHttpServerTransport] [test-cluster-0] publish_address {127.0.0.1:59796}, bound_addresses {[::1]:59795}, {127.0.0.1:59796}\r\n[2024-06-21T14:03:51,165][INFO ][o.e.n.Node               ] [test-cluster-0] started {test-cluster-0}{H7cMutpyT02cCctNxO4BKA}{38JRkdpbQcGHJu-VAuUa3A}{test-cluster-0}{127.0.0.1}{127.0.0.1:59794}{cdfhilmrstw}{8.15.0}{7000099-8508000}{testattr=test, xpack.installed=true}\r\n[2024-06-21T14:03:51,289][INFO ][o.e.c.s.MasterService    ] [test-cluster-0] elected-as-master ([1] nodes joined in term 1)[_FINISH_ELECTION_, {test-cluster-0}{H7cMutpyT02cCctNxO4BKA}{38JRkdpbQcGHJu-VAuUa3A}{test-cluster-0}{127.0.0.1}{127.0.0.1:59794}{cdfhilmrstw}{8.15.0}{7000099-8508000} completing election], term: 1, version: 1, delta: master node changed {previous [], current [{test-cluster-0}{H7cMutpyT02cCctNxO4BKA}{38JRkdpbQcGHJu-VAuUa3A}{test-cluster-0}{127.0.0.1}{127.0.0.1:59794}{cdfhilmrstw}{8.15.0}{7000099-8508000}]}\r\n[2024-06-21T14:03:51,339][INFO ][o.e.c.c.CoordinationState] [test-cluster-0] cluster UUID set to [iPH2DCsNQ5-fh7-dpqfylg]\r\n[2024-06-21T14:03:51,339][INFO ][o.e.c.u.FeatureFlag      ] [test-cluster-0] The current build is a snapshot, feature flag [failure_store] is enabled\r\n[2024-06-21T14:03:51,379][INFO ][o.e.c.s.ClusterApplierService] [test-cluster-0] master node changed {previous [], current [{test-cluster-0}{H7cMutpyT02cCctNxO4BKA}{38JRkdpbQcGHJu-VAuUa3A}{test-cluster-0}{127.0.0.1}{127.0.0.1:59794}{cdfhilmrstw}{8.15.0}{7000099-8508000}]}, term: 1, version: 1, reason: Publication{term=1, version=1}\r\n[2024-06-21T14:03:51,432][INFO ][o.e.c.c.NodeJoinExecutor ] [test-cluster-0] node-join: [{test-cluster-0}{H7cMutpyT02cCctNxO4BKA}{38JRkdpbQcGHJu-VAuUa3A}{test-cluster-0}{127.0.0.1}{127.0.0.1:59794}{cdfhilmrstw}{8.15.0}{7000099-8508000}] with reason [completing election]\r\n[2024-06-21T14:03:51,464][INFO ][o.e.c.f.AbstractFileWatchingService] [test-cluster-0] starting file watcher ...\r\n[2024-06-21T14:03:51,471][INFO ][o.e.c.f.AbstractFileWatchingService] [test-cluster-0] file settings service up and running [tid=67]\r\n[2024-06-21T14:03:51,472][INFO ][o.e.r.s.FileSettingsService] [test-cluster-0] setting file [C:\\bk\\test\\external-modules\\jvm-crash\\build\\testrun\\javaRestTest\\temp\\test-cluster3175310756303214425\\test-cluster-0\\config\\operator\\settings.json] not found, initializing [file_settings] as empty\r\n[2024-06-21T14:03:51,473][INFO ][o.e.g.GatewayService     ] [test-cluster-0] recovered [0] indices into cluster_state\r\n[2024-06-21T14:03:51,583][INFO ][o.e.h.n.s.HealthNodeTaskExecutor] [test-cluster-0] Node [{test-cluster-0}{H7cMutpyT02cCctNxO4BKA}] is selected as the current health node.\r\n[2024-06-21T14:03:51,785][INFO ][o.e.l.ClusterStateLicenseService] [test-cluster-0] license [cffbba22-d938-4877-965f-1457971c195a] mode [basic] - valid\r\n"
```

**Mute Reasons:**
- [main] 3 failures in test testJvmCrash (4.7% fail rate)

**Note:**
This issue was created using new, in-development muting automation. Please mind any issues.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -73,9 +73,6 @@ tests:
 - class: org.elasticsearch.xpack.transform.transforms.TransformIndexerTests
   method: testMaxPageSearchSizeIsResetToConfiguredValue
   issue: https://github.com/elastic/elasticsearch/issues/109844
-- class: "org.elasticsearch.test.jvm_crash.JvmCrashIT"
-  issue: "https://github.com/elastic/elasticsearch/issues/110052"
-  method: "testJvmCrash"
 
 # Examples:
 #

--- a/test/external-modules/jvm-crash/src/javaRestTest/java/org/elasticsearch/test/jvm_crash/JvmCrashIT.java
+++ b/test/external-modules/jvm-crash/src/javaRestTest/java/org/elasticsearch/test/jvm_crash/JvmCrashIT.java
@@ -50,7 +50,7 @@ public class JvmCrashIT extends ESRestTestCase {
 
     @BeforeClass
     public static void dontRunWindows() {
-        assumeFalse("Stdout redirects don't work on windows", OS.current() == OS.WINDOWS);
+        assumeFalse("JVM crash log doesn't go to stdout on windows", OS.current() == OS.WINDOWS);
     }
 
     private static class StdOutCatchingClusterBuilder extends AbstractLocalClusterSpecBuilder<ElasticsearchCluster> {

--- a/test/external-modules/jvm-crash/src/javaRestTest/java/org/elasticsearch/test/jvm_crash/JvmCrashIT.java
+++ b/test/external-modules/jvm-crash/src/javaRestTest/java/org/elasticsearch/test/jvm_crash/JvmCrashIT.java
@@ -22,9 +22,11 @@ import org.elasticsearch.test.cluster.local.distribution.DistributionType;
 import org.elasticsearch.test.cluster.local.distribution.LocalDistributionResolver;
 import org.elasticsearch.test.cluster.local.distribution.ReleasedDistributionResolver;
 import org.elasticsearch.test.cluster.local.distribution.SnapshotDistributionResolver;
+import org.elasticsearch.test.cluster.util.OS;
 import org.elasticsearch.test.rest.ESRestTestCase;
 import org.hamcrest.Matcher;
 import org.junit.AfterClass;
+import org.junit.BeforeClass;
 import org.junit.ClassRule;
 
 import java.io.BufferedReader;
@@ -45,6 +47,11 @@ import static org.hamcrest.Matchers.matchesRegex;
 import static org.hamcrest.Matchers.not;
 
 public class JvmCrashIT extends ESRestTestCase {
+
+    @BeforeClass
+    public static void dontRunWindows() {
+        assumeFalse("Stdout redirects don't work on windows", OS.current() == OS.WINDOWS);
+    }
 
     private static class StdOutCatchingClusterBuilder extends AbstractLocalClusterSpecBuilder<ElasticsearchCluster> {
 

